src/Compilers/Core/Portable/WellKnownTypes.cs

```diff
@@ -252,12 +253,26 @@ internal enum WellKnownType
         System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6,
         System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7,
 
+        ExtendedType = 255, // Not a real type, just a marker for types above 255 and strictly below 512
+
+        System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest,
+
         System_IFormatProvider,
 
         Available,
         Last = Available - 1,
     }
 
+    internal static class WellKnownTypeExtensions
+    {
+        public static byte ExtensionByte(this WellKnownType type)
```

Comment:
It looks like, given the other feedback, this method is not going to be used. Consider removing.